libs/s25main/mapGenerator/RandomUtility.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2005 - 2017 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include "random/XorShift.h"
+#include "gameTypes/MapCoordinates.h"
+#include <vector>
+
+namespace rttr { namespace mapGenerator {
+
+    using UsedRNG = XorShift;
```

Comment:
I think you could just use one of the std:: PRNGs here, or what was the reason to using that here?

Reply:
I just used what it was before. The original reason not to use std random number generators was that they could produce different numbers depending on operating system. So I used the existing class instead, as far as I remember.